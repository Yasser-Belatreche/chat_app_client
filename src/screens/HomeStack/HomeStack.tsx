import React from 'react';
import {
  createNativeStackNavigator,
  NativeStackNavigationOptions,
} from '@react-navigation/native-stack';

// types
import {HomeStackParamsList} from './HomeStack.types';
import {MainStackScreenProps} from '../MainStack.types';

// screens
import {Home} from './screens/Home/Home';

interface Props extends MainStackScreenProps<'HomeStack'> {}

const Stack = createNativeStackNavigator<HomeStackParamsList>();

const HomeStack: React.FC<Props> = () => {
  const screensOptions: NativeStackNavigationOptions = {
    headerShown: false,
  };

  return (
    <Stack.Navigator screenOptions={screensOptions}>
      <Stack.Screen name="Home" component={Home} />
    </Stack.Navigator>
  );
};

export {HomeStack};
