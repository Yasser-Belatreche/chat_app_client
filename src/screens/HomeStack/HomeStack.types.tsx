import {NativeStackScreenProps} from '@react-navigation/native-stack';

export type HomeStackParamsList = {
  Home: undefined;
};

export type HomeStackScreenProps<
  CurrentScreen extends keyof HomeStackParamsList,
> = NativeStackScreenProps<HomeStackParamsList, CurrentScreen>;
