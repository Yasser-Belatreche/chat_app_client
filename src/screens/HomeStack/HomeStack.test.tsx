import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {NavigationContainer} from '@react-navigation/native';

// import {fireEvent, render} from '@testing-library/react-native';

import {HomeStack} from './HomeStack';

describe('HomeStack', () => {
  const props: any = {};

  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <NavigationContainer>
          <HomeStack {...props} />
        </NavigationContainer>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
