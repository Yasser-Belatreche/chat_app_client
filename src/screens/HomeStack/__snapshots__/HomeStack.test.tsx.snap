// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeStack should renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        hideShadow={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3b4ba5",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 30,
                "fontWeight": "700",
                "marginBottom": 10,
              }
            }
          >
            Happy Coding ✌️✌️
          </Text>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
            <Text>
              Made By 
            </Text>
            <Text
              onPress={[Function]}
              style={
                Object {
                  "textDecorationLine": "underline",
                }
              }
            >
              Yasser
            </Text>
          </Text>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
